clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3040,6 +3040,11 @@ bool Sema::checkTargetVersionAttr(SourceLocation LiteralLoc, Decl *D,
   enum FirstParam { Unsupported };
   enum SecondParam { None };
   enum ThirdParam { Target, TargetClones, TargetVersion };
+
+  if (!Context.getTargetInfo().getTriple().isRISCV() &&
+      !Context.getTargetInfo().getTriple().isAArch64())
+    return Diag(LiteralLoc, diag::err_target_version_unsupported);
```

Comment:
A better way to handle this is to use `TargetSpecificAttr` for `TargetVersion` in `Attr.td` (instead of `InheritableAttr`). That should automatically handle performing this check and you won't need to add a custom diagnostic.

Reply:
Great idea. This patch now uses the TargetSpecificAttr to enable target_version for AArch64 and RISC-V.